lib/Doctrine/DBAL/Driver/PDOSqlsrv/Connection.php

```diff
@@ -40,6 +40,22 @@ public function __construct($dsn, $user = null, $password = null, array $options
     /**
      * {@inheritDoc}
      */
+    public function lastInsertId($name = null)
+    {
+        if (null === $name) {
```

Comment:
@Ocramius the problem is the interface does not specify such case. Neither does PDO's interface. No exception, not non-string return value specified.